Framework/Crystal/src/PredictPeaks.cpp

```diff
@@ -489,10 +492,58 @@ void PredictPeaks::calculateQAndAddToOutput(const V3D &hkl,
   if (m_sfCalculator) {
     p.setIntensity(m_sfCalculator->getFSquared(hkl));
   }
-
+  int edge = this->getProperty("EdgePixels");
```

Comment:
Maybe this could be moved a bit further up - before the matrix is set and before potential calculation of the structure factor, that would save some more calculations for the peaks affected by this check.